### PR TITLE
Added chronicler support in the vault

### PR DESCRIPTION
**Pull request by MartinSchmidt**
Adds support for the Vault the register a claim intent at a chronicler for an area before making the allocation and claim on the registry.

### CODE REVIEW

File: src/ProjectOrigin.Vault/Activities/AllocateActivity.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        try
+        {
+            var cons = await _unitOfWork.CertificateRepository.GetWalletSlice(context.Arguments.ConsumptionSliceId);
+            var prod = await _unitOfWork.CertificateRepository.GetWalletSlice(context.Arguments.ProductionSliceId);
+
+            byte[]? chroniclerSignature = context.Arguments.ChroniclerRequestId is not null
+                ? context.GetVariable<byte[]>(context.Arguments.ChroniclerRequestId.Value.ToString())
+                : null;
+
+            var allocatedEvent = CreateAllocatedEvent(context.Arguments.AllocationId, cons, prod, chroniclerSignature);
```
Comment (tnickelsen) [inline src/ProjectOrigin.Vault/Activities/AllocateActivity.cs:54]: I thought allocation was just something in the wallet but we're sending allocation to the registry now. Why is that?

File: protos/chronicler.proto [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+message ClaimIntentRequest {
+    // The id of the certificate to claim
+    project_origin.common.v1.FederatedStreamId certificate_id = 1;
+
+    // The quantity of the certificate to claim
+    int32 quantity = 2;
+
+    // The random value to be used in the claim
+    bytes random_r = 3;
```
Comment (tnickelsen) [inline protos/chronicler.proto:20]: why is the r value needed when you have the quantity?

Comment (MartinSchmidt) [reply protos/chronicler.proto:20]: To be able to calculate the commitment and verify it.

File: src/ProjectOrigin.Vault/Activities/AllocateActivity.cs (hunk 1 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.Vault/Activities/AllocateActivity.cs:54]: Always have been ;)

Comment (tnickelsen) [reply src/ProjectOrigin.Vault/Activities/AllocateActivity.cs:54]: What are these allocation events? Maybe we should take a call

**Review by tnickelsen — APPROVED** [review]

**Pull request merged**